src/components/pagination/index.tsx

```diff
@@ -90,14 +98,17 @@ const Pagination: FC<PropsWithChildren<Props>> = (props) => {
       marginTop={marginTop}
       marginBottom={marginBottom}
       __css={paginationContainer}
+      data-testid="pagination"
     >
       <Show above="xs">
         <PaginationButton
-          isDisabled={isFirstPage}
+          isDisabled={!infinite && isFirstPage}
           onClick={onPrevious}
           ariaLabel="previous page"
```

Comment:
isn't this now a bit misleading since it's not really a page but a slide in a carousel?